src/migrateCalculatedMeasures.ts

```diff
@@ -1,41 +1,89 @@
 import { ContentRecord } from "@activeviam/activeui-sdk";
-import {
-  migrateCalculatedMeasure,
-  LegacyCalculatedMeasure,
-} from "./migrateCalculatedMeasure";
+import { getCalculatedMeasures } from "./getCalculatedMeasures";
+import { generateId } from "./generateId";
+
+export interface LegacyCalculatedMeasure {
+  expression: string;
+  formatStringExpression: string;
+  uniqueName: string;
+  owners: string[];
+  readers: string[];
+  isDirectory: boolean;
+}
 
 /**
  * Returns the converted content record tree for the calculted measures, ready to be used in ActiveUI 5.
  */
-export function migrateCalculatedMeasures(
-  legacyCalculatedMeasures: LegacyCalculatedMeasure[],
-  calculatedMeasuresFolder: ContentRecord
-): ContentRecord {
-  const migratedCalculatedMeasures = legacyCalculatedMeasures.map(
-    (legacyMeasure) => migrateCalculatedMeasure(legacyMeasure)
-  );
+export async function migrateCalculatedMeasures(
+  legacyPivotFolder: ContentRecord
+): Promise<ContentRecord> {
+  const entitlements = legacyPivotFolder?.children?.entitlements;
+  const calculatedMeasuresFolder = entitlements?.children?.cm!;
```

Comment:
Are you sure that `cm` is necessarily defined in there if there are no calculated measures ?
I would handle the undefined case just out safety: you can return `{}` early in that case.